jme3-core/src/main/java/com/jme3/shader/DefineList.java

```diff
@@ -142,21 +143,27 @@ public int getInt(int id) {
 
     @Override
     public int hashCode() {
-        return (int) ((isSet >> 32) ^ isSet);
+        return isSet.hashCode();
     }
 
     @Override
-    public boolean equals(Object other) {
-        DefineList o = (DefineList) other;
-        if (isSet == o.isSet) {
-            for (int i = 0; i < values.length; i++) {
-                if (values[i] != o.values[i]) {
-                    return false;
-                }
-            }
+    public boolean equals(Object object) {
+        if(this == object ) {
             return true;
         }
-        return false;
+        if(object ==  null || object.getClass() != getClass()) {
+            return false;
+        }
+        DefineList otherDefineList = (DefineList) object;
+        if (!isSet.equals(otherDefineList.isSet)) {
+            return false;
+        }            
+        for (int i = 0; i < values.length; i++) {
```

Comment:
What if the `values` arrays are of different lengths? In that case, I believe some elements won't be compared!

Reply:
that is a good point, i think that would also make it possible to throw an array out of bounds exception in the case that they are different sizes.

Although I also think the code right above the values check `if (!isSet.equals(otherDefineList.isSet)`) { that is checking for equal bitSets should already cause the equals() method to return false if the values array have different sizes (since values and isSet both appear to get initialized with the same size in the class' constructor)

But it could still be a good idea to add a length check for the size of the values array and return false if they are not equal lengths prior to looping through the arrays, just to be on the safe side.

Reply:
i added the changes, although unfortunately that did not have any effect on the failing tests.